index.js

```diff
@@ -1,5 +1,111 @@
 const battleship = () => {
-  return 'The winner is...?'
+
+  //step 1
+
+  //objects of players(works)
+  let player1 = {
+    name: prompt("Player 1, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+    }
+
+    let player2 = {
+    name: prompt("Player 2, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+
+  }
+
+
+  //step 2(notice number of placed ships are not always consistent when ran. sometimes it shows 4 other times 3 maybe a two. not sure why. if you see what's wrong, please give me feedback. thanks!
+
+  //setup of ships
+
+    let ships;
+    let xPos;
+    let yPos;
+      for (i=0; i<player1.shipCount; i++){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player1.gameBoard[xPos][yPos])!==1){
+        player1.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+      }
+
+      for (i=0; i<player2.shipCount; i++){
```

Comment:
This loop needs to run not only only 4 times but as many times as is needed for each player to get 4 ships on their board (you might get a random x,y coordinate for a space that already has a ship, so the loop might need to run more than 4 times)